src/Product/SearchProvider.php

```diff
@@ -213,13 +232,25 @@ public function runQuery(
     private function generateCacheKeyForQuery(ProductSearchQuery $query, array $facetedSearchFilters)
     {
         $context = $this->module->getContext();
+
+        $filterKey = $query->getQueryType();
+        if ($query->getQueryType() == 'category') {
+            $filterKey .= $query->getIdCategory();
+        } elseif ($query->getQueryType() == 'manufacturer') {
+            $filterKey .= $query->getIdManufacturer();
+        } elseif ($query->getQueryType() == 'supplier') {
+            $filterKey .= $query->getIdSupplier();
+        } elseif ($query->getQueryType() == 'search') {
+            $filterKey .= $query->getSearchString() . $query->getSearchTag();
```

Comment:
Search is not cachable anyway, right? I don't know exactly how it works, but generating a cache for every single search string would be overkill.

Reply:
Fixed